src/reconcile/rec_visibility.c

```diff
@@ -571,6 +674,20 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
     if (has_newer_updates)
         r->leave_dirty = true;
 
+    onpage_upd = upd_select->upd != NULL && upd_select->upd->type == WT_UPDATE_TOMBSTONE ?
+      NULL :
+      upd_select->upd;
+
+    /* Check the update chain for conditions that could prevent it's eviction. */
+    WT_ERR(__rec_validate_upd_chain(session, r, onpage_upd, select_tw, vpack));
+
+    /*
+     * Fixup any out of order timestamps, assert that checkpoint isn't running if we're in eviction.
+     */
+    if (__timestamp_out_of_order_fix(session, select_tw) &&
+      F_ISSET(S2C(session), WT_CONN_HS_OPEN) && F_ISSET(r, WT_REC_CHECKPOINT_RUNNING))
```

Comment:
The order surprises me here - wouldn't we want to panic before fixing up out of order timestamps?

Reply:
This should turn into an assert not panic.

Reply:
This will turn into a WT_ERR(EBUSY) an an assert, the reason I went with a panic was that in theory its impossible. As the `__rec_validate_upd_chain` call should always catch this case, which of course an assert will achieve too in diagnostic builds.